### PR TITLE
Training overhaul

### PR DESCRIPTION
**Pull request by tekn0ir**

### CODE REVIEW

**Review by GNEILTek — DISMISSED** [review]

File: notebooks/train_yolov7.ipynb [hunk 1]
```diff
@@ -172,118 +178,79 @@
...
-    "        # output = namedtuple('YOLOArgs', ['data_file', 'config_file', 'transfer_weights_file', 'hyp_file', 'names_file'])\n",
-    "        # return output(\"\", \"\", \"\", \"\", \"\")\n",
-    "        return\n",
+    "    if len(tasks) == 0:\n",
+    "        print(f'NO TASKS FOUND FOR {project}')\n",
+    "        exit(0)\n",
     "\n",
+    "    ind_list = list(range(len(tasks)))\n",
+    "    shuffle(ind_list)\n",
     "    test_frac = 1.0 - train_frac - validate_frac\n",
```
Comment (fdarvas) [inline notebooks/train_yolov7.ipynb:205]: do we need a test/validation split here? Unless we adjust model hyperparameters and fine tune those, we wouldnt need a final test set?

**Review by fdarvas — DISMISSED** [review]

File: notebooks/train_yolov7.ipynb (hunk 1 above)
Comment (tekn0ir) [reply notebooks/train_yolov7.ipynb:205]: Those arguments are parameterized and can be set arbitrarily for each training run.
